config/transformers/shadowToCss.test.ts

```diff
@@ -0,0 +1,171 @@
+    expect(input.map(item => shadowToCss.transformer(item))).toStrictEqual(expectedOutput)
+  })
+
+  it('throws an error when required values are missing', () => {
+    // missing blur
+    expect(() =>
+      shadowToCss.transformer(
+        getMockToken({
+          value: {
+            color: '#000000',
+            offsetX: '2px',
+            offsetY: '2px',
+            blur: '1px'
+          }
+        })
+      )
+    ).toThrowError()
+
+    // missing spread
+    expect(() =>
+      shadowToCss.transformer(
+        getMockToken({
+          value: {
+            color: '#000000',
+            offsetX: '2px',
+            offsetY: '2px',
+            blur: '1px'
+          }
+        })
+      )
+    ).toThrowError()
+
+    // missing offsets
+    expect(() =>
+      shadowToCss.transformer(
+        getMockToken({
+          value: {
+            color: '#000000',
+            offsetX: '2px',
+            spread: '0px',
+            blur: '1px'
+          }
+        })
+      )
+    ).toThrowError()
+
+    expect(() =>
+      shadowToCss.transformer(
+        getMockToken({
+          value: {
+            color: '#000000',
+            offsetY: '2px',
+            spread: '0px',
+            blur: '1px'
+          }
+        })
+      )
+    ).toThrowError()
+    // missing color
+    expect(() =>
+      shadowToCss.transformer(
+        getMockToken({
+          value: {
+            offsetX: '0px',
+            offsetY: '2px',
+            spread: '0px',
+            blur: '1px'
+          }
+        })
+      )
+    ).toThrowError()
+  })
+
+  it('transforms `shadow` token alpha value to css shadow string', () => {
+    const input = [
+      getMockToken({
+        value: {
+          color: '#000000',
+          offsetX: '0px',
+          offsetY: '2px',
+          blur: '1px',
+          spread: '0',
+          alpha: 0.5
+        }
+      }),
+      getMockToken({
+        value: {
+          color: '#22222266',
+          offsetX: '0px',
+          offsetY: '2px',
+          blur: '1px',
+          spread: '0',
+          alpha: 0.5
+        }
+      })
+    ]
+    const expectedOutput = ['0px 2px 1px 0 #00000080', '0px 2px 1px 0 #22222280']
+    expect(input.map(item => shadowToCss.transformer(item))).toStrictEqual(expectedOutput)
+  })
+
+  it('transforms multi-layer `shadow` token to css shadow string', () => {
+    const input = [
+      getMockToken({
+        value: [
+          {
+            color: '#000000',
+            offsetX: '0px',
+            offsetY: '2px',
+            blur: '1px',
+            spread: '0',
+            alpha: 0.5
+          },
+          {
+            color: '#22222266',
+            offsetX: '0px',
+            offsetY: '8px',
+            blur: '16px',
+            spread: '0',
+            alpha: 0.2
+          }
+        ]
+      })
+    ]
+    const expectedOutput = ['0px 2px 1px 0 #00000080, 0px 8px 16px 0 #22222233']
```

Comment:
I would expect the output to be `0px 2px 1px 0 #00000080, 0px 8px 16px 0 #22222233` not an array.

The mapping of multiple input objects is a side-effect and isn't the responsibility of `shadowToCss`. Is it possible to keep this test `pure` and only test the main fn? Some maintainers - like myself - may use the tests as a way to understand what the functions do regarding I/O, but it's not immediately clear here.

Reply:
The array is due to the way the test is written. Since we are only testing one value it is not needed and I have updated it.